File: Dante1/Module1.04/Zadanie4.13/main.c
#include <stdio.h>

int main()
{
    int error;
    int array[3000];
    int size_of_array = 0;

    printf("Enter array: ");

    for(int i = 0; i < 3000; i++)
    {
        error = scanf("%i", &array[i]);
        if(error != 1)
        {
            printf("Incorrect input");
            return 1;
        }
        if(array[i] == 0)
        {
            break;
        }
        size_of_array++;
    }
    if(size_of_array > 1000)
    {
        size_of_array = 1000;
    }

    if (size_of_array == 0)
    {
        printf("\n0\n0\n");
        return 0;
    }

    if (size_of_array == 1)
    {
        printf("\n0\n0");
        return 0;
    }

    int max = 0, counter = 0, temp = 1;
    int some_shit[1000] = {0}, storage_for_index[1000] = {0};

    for (int i = 1; i < size_of_array; ++i)
    {
        if (array[i] > array[i - 1])
        {
            temp++;
        }
        else
        {
            if (temp > max)
            {
                max = temp;
                counter = 1;
                some_shit[0] = temp;
                storage_for_index[0] = i - temp;
            }
            else if (temp == max)
            {
                some_shit[counter] = temp;
                storage_for_index[counter] = i - temp;
                counter++;
            }
            temp = 1;
        }
    }

    if (temp > max)
    {
        max = temp;
        counter = 1;
        some_shit[0] = temp;
        storage_for_index[0] = size_of_array - temp;
    }
    else if (temp == max)
    {
        some_shit[counter] = temp;
        storage_for_index[counter] = size_of_array - temp;
        counter++;
    }
    if (max < 2)
    {
        printf("\n0\n0\n");
        return 0;
    }

    printf("\n%i\n%i\n", max, counter);
    for (int i = 0; i < counter; i++)
    {
        printf("%i -", storage_for_index[i]);
        for (int j = 0; j < some_shit[i]; j++)
        {
            printf(" %i", array[storage_for_index[i] + j]);
        }
        printf("\n");
    }

    return 0;
}

File: Dante1/Module1.04/Zadanie4.13/main.rs
fn main()
{
    let mut array:[i32; 3000] = [0; 3000];
    let mut size_of_array: usize = 0;
    
    println!("Enter: ");
    let mut input: String = String::new();
    
    for _i in 0 .. 3000
    {
        input.clear();
        std::io::stdin().read_line(&mut input).expect("Error");
        let number: i32 = input.trim().parse().expect("Error");
        
        if number == 0
        {
            break;
        }
        
        array[size_of_array] = number;
        size_of_array += 1;
    }
    
    if size_of_array > 1000
    {
        size_of_array = 1000;
    }
    
    if size_of_array == 0 || size_of_array == 1
    {
        println!("{}\n{}\n", 0 ,0);
        return;
    }
    
    let mut max: i32 = i32::MIN;
    let mut counter: i32 = 0;
    let mut temp: i32 = 1;
    
    let mut some_shit:[i32; 1000] = [0; 1000];
    let mut storage_for_index:[i32; 1000] = [0; 1000];
    
    for i in 1 .. size_of_array
    {
        if array[i] > array[i - 1]
        {
            temp += 1;
        }
        else
        {
            if temp > max
            {
                max = temp;
                
                counter = 1;
                some_shit[0] = temp;
                storage_for_index[0] = i as i32 - temp;
            }
            else if temp == max
            {
                some_shit[counter as usize] = temp;
                storage_for_index[counter as usize] = i as i32 - temp;
                counter += 1;
            }
            temp = 1;
        }
    }

    if temp > max
    {
        max = temp;
        counter = 1;
        some_shit[0] = temp;
        storage_for_index[0] = size_of_array as i32 - temp;
    }
    else if temp == max
    {
        some_shit[counter as usize] = temp;
        storage_for_index[counter as usize] = size_of_array as i32 - temp;
        counter += 1;
    }
    if max < 2
    {
        println!("\n0\n0\n");
        return;
    }

    println!("{}\n{}", max, counter);
    for i in 0 .. counter
    {
        print!("{} -", storage_for_index[i as usize]);
        for j in 0 .. some_shit[i as usize]
        {
            print!(" {}", array[(storage_for_index[i as usize] + j) as usize]);
        }
        println!();
    }
    
    return;
}
